ApplicationUnitTests/Commands/Identities/DeleteIdentityHandlerTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Moq;
+using Identity_Application.Commands.Identities;
+using Identity_Application.Interfaces.Repository;
+using Identity_Domain.Entities.Base;
+
+namespace ApplicationUnitTests.Commands.Identities;
+
+public class DeleteIdentityHandlerTests
+{
+    private readonly Mock<IGenericRepository<Identity>> _mockIdentityRepository;
+    private readonly DeleteIdentityHandler _handler;
+
+    public DeleteIdentityHandlerTests()
+    {
+        _mockIdentityRepository = new Mock<IGenericRepository<Identity>>();
+        _handler = new DeleteIdentityHandler(_mockIdentityRepository.Object);
+    }
+
+    [Fact]
+    public async Task Handle_ValidCommand_ShouldDeleteIdentity()
+    {
+        // Arrange
+        var identityId = 1;
+        var deleteIdentityCommand = new DeleteIdentityCommand(identityId);
+
+        _mockIdentityRepository.Setup(repo => repo
+                               .DeleteAsync(identityId));
+
+        // Act
+        await _handler.Handle(deleteIdentityCommand, new CancellationToken());
+
+        // Assert
+        _mockIdentityRepository.Verify(repo => repo
+                               .DeleteAsync(identityId), Times.Once);
+    }
+
+    [Fact]
+    public async Task Handle_InvalidCommand_ThrowsException()
+    {
+        // Arrange
+        var invalidIdentityId = -1;
```

Comment:
I see you use it a multiple time  and `validIdentityId` and `identityId`
maybe we can move it to constant and reuse in each unit test class?( just a thought)

Reply:
It is used a lot of times, but they both can and should be unique in each test)